apps/web/src/middlewares/client-id-middleware.ts

```diff
@@ -0,0 +1,23 @@
+/* eslint-disable no-param-reassign */
+import { NextFetchEvent, NextResponse } from 'next/server'
+import { v4 } from 'uuid'
+import { ONE_YEAR_MILLISECONDS } from './constants'
+import { ExtendedNextReq, MiddlewareFactory, NextMiddleware } from './types'
+
+const clientCookieOptions = { httpOnly: true, secure: true, maxAge: ONE_YEAR_MILLISECONDS }
+
+export const withClientId: MiddlewareFactory = (next: NextMiddleware) => {
+  return async (request: ExtendedNextReq, _next: NextFetchEvent) => {
+    const randomId = v4()
+
+    const clientIdCookie = request.cookies.get('p_client_id')
+    const clientId = clientIdCookie ? clientIdCookie.value : randomId
```

Comment:
only generate a new one if necessary
```suggestion
    const clientIdCookie = request.cookies.get('p_client_id')
    const clientId = clientIdCookie ? clientIdCookie.value : v4()
```